packages/synthetics-sdk-broken-links/test/unit/broken_links.spec.ts

```diff
@@ -157,4 +158,130 @@ describe('GCM Synthetics Broken Links Core Functionality', async () => {
       expect(result.retriesRemaining).to.equal(2);
     });
   });
+  describe('retrieveLinksFromPage', async () => {
```

Comment:
A few other scenarios I can think of -- what happens if await page.evaluate throws an error?

Reply:
I don't think it should ever throw an error. One question would be what do we do if no links are scraped, users input for `query_selector_all` is wrong. I think it is fine to do nothing, this should be evident to users but worth checking with product. wdyt?

Reply:
https://screenshot.googleplex.com/6hYY4pETDoWjMie - always expect the unexpected. 

We should keep in mind that anything that awaits should have a catch around it _somewhere_ that ultimately catches unexpected errors.

Reply:
I've added a try-catch block in `runBrokenLinks` with a TODO in the catch block to throw a generic result with relevant message